src/main/java/com/mraof/minestuck/item/loot/functions/SetSburbCodeFragments.java

```diff
@@ -0,0 +1,72 @@
+package com.mraof.minestuck.item.loot.functions;
+
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonSerializationContext;
+import com.mraof.minestuck.item.IncompleteSburbCodeItem;
+import com.mraof.minestuck.item.loot.MSLootTables;
+import com.mraof.minestuck.util.MSTags;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.storage.loot.LootContext;
+import net.minecraft.world.level.storage.loot.functions.LootItemConditionalFunction;
+import net.minecraft.world.level.storage.loot.functions.LootItemFunctionType;
+import net.minecraft.world.level.storage.loot.predicates.LootItemCondition;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+public class SetSburbCodeFragments extends LootItemConditionalFunction
+{
+	public SetSburbCodeFragments(LootItemCondition[] conditionsIn)
+	{
+		super(conditionsIn);
+	}
+	
+	@Override
+	public LootItemFunctionType getType()
+	{
+		return MSLootTables.SET_SBURB_CODE_FRAGMENT_FUNCTION.get();
+	}
+	
+	@Override
+	protected ItemStack run(ItemStack stack, LootContext context)
+	{
+		return IncompleteSburbCodeItem.setRecordedInfo(stack, getRandomList(context.getRandom()));
+	}
+	
+	private List<Block> getRandomList(Random random)
+	{
+		List<Block> completeList = MSTags.getBlocksFromTag(MSTags.Blocks.GREEN_HIEROGLYPHS);
+		List<Block> pickedList = new ArrayList<>();
+		if(!completeList.isEmpty())
+		{
+			int numberOfIterations = Math.max(1, random.nextInt(Math.min(3, completeList.size())));
+			
+			for(int iterate = 0; iterate < numberOfIterations; iterate++) //up to two runs at filling the book with another fragment of code
+			{
+				pickedList.add(completeList.get(random.nextInt(completeList.size()))); //random element from the full list of blocks
```

Comment:
Seems like this could end up adding the same block twice (or more) to the list. This doesn't seem intended.